### PR TITLE
test: clean up v31 disabled test coverage

### PR DESCRIPTION
**Pull request by valera-grinenko-ai**
## Summary
- Remove obsolete commented/skipped v31 test scaffolding and delete skip-only test wrappers.
- Restore the missing-upgrade-log TypeScript test and Bridgehub `removeChainTypeManager` non-owner coverage as active assertions.
- Document every removed test case, remaining intentional skip, replacement coverage, and validation status in `v31-test-followups.md`.

## Validation
- `cd l1-contracts && forge test --ffi --match-path 'test/foundry/l1/unit/concrete/Bridgehub/experimental_bridge.t.sol' --match-test 'test_removeChainTypeManager_cannotBeCalledByRandomAddress'` - passed 1 test
- `cd da-contracts && yarn test:foundry` - passed 8 tests
- `cd l1-contracts && yarn test:foundry` - passed 2561 tests
- `cd system-contracts && PATH=/tmp/foundry-zksync-v0.0.30:$PATH yarn build:foundry` - passed after using the repo-specified foundry-zksync v0.0.30 binary
- `cd l1-contracts && PATH=/tmp/foundry-zksync-v0.0.30:$PATH yarn test:zkfoundry` - failed 12 L2 integration suites in `setUp()` with `EvmError: Revert` / `Invalid opcode, Not enough gas`; representative `WETH.t.sol` repro with higher gas limit failed the same way after `Deploying L2 contracts`
- `yarn lint:check` - passed
- skipped/commented-test scans over `l1-contracts`, `system-contracts`, and `da-contracts`

### CODE REVIEW

File: system-contracts/test/EvmEmulation.spec.ts [hunk 1]
```diff
@@ -35,7 +35,8 @@ describe("EvmEmulation tests", function () {
     expect(testValue).to.be.eq(101);
   });
 
-  // TODO: anvil-zksync uses old EVM emulator
+  // Disabled until anvil-zksync ships the EVM emulator version that supports BLOBBASEFEE.
+  // Re-enable when the runner executes the opcode and validates the expected blob base fee.
```
Comment (jcsec-security) [inline system-contracts/test/EvmEmulation.spec.ts]: - Are we tracking this pending re-enabling somewhere?
- Is it actually planned to support BLOBBASEFEE in anvil-zksync?

File: v31-test-followups.md [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+- Risk while disabled: emulator support drift for `BLOBBASEFEE` can be missed by the system-contracts suite.
+
+### `system-contracts/test/EvmEmulation.spec.ts` - `Can use BLOBHASH opcode`
+
+- Reason disabled: anvil-zksync currently uses an EVM emulator that does not execute the Cancun `BLOBHASH` opcode in this harness.
+- Action needed: update the test runner/emulator dependency, then assert the returned blob hash through the deployed bytecode path.
+- Re-enable criteria: the raw-bytecode contract deploys and calls successfully under the system-contracts runner, and the test asserts the expected blob hash value.
+- Risk while disabled: emulator support drift for `BLOBHASH` can be missed by the system-contracts suite.
+
+### `l1-contracts/package.json` - `UpgradeTestv31_Remote.test_DefaultUpgrade_MainnetFork`
```
Comment (jcsec-security) [inline v31-test-followups.md]: This is just expected in out setup, right?

File: l1-contracts/test/foundry/l1/unit/concrete/Bridgehub/experimental_bridge.t.sol [hunk 3]
```diff
@@ -448,82 +448,30 @@ contract ExperimentalBridgeTest is Test {
...
+        vm.assume(_randomAddressWithoutTheCorrectInterface != address(0));
+        vm.assume(_randomCaller != bridgeOwner);
+
+        vm.prank(_randomCaller);
+        vm.expectRevert(bytes("Ownable: caller is not the owner"));
+        bridgehub.removeChainTypeManager(_randomAddressWithoutTheCorrectInterface);
+
+        vm.prank(bridgeOwner);
+        bridgehub.addChainTypeManager(_randomAddressWithoutTheCorrectInterface);
+        assertTrue(bridgehub.chainTypeManagerIsRegistered(_randomAddressWithoutTheCorrectInterface));
```
Comment (jcsec-security) [inline l1-contracts/test/foundry/l1/unit/concrete/Bridgehub/experimental_bridge.t.sol]: The specific naming `_randomAddressWithoutTheCorrectInterface` made me think of the comment in `addChainTypeManager` ["Chain Type Manager can be any contract with the appropriate interface/functionality" ](https://github.com/matter-labs/era-contracts/blob/draft-v31/l1-contracts/contracts/core/bridgehub/BridgehubBase.sol#L226-L228). 

Do we consider that it is not worth adding an interface implementation check before adding the CTM to bridgehub? From reading just the test I would have expected `_randomAddressWithoutTheCorrectInterface` to be rejected

File: l1-contracts/test/foundry/l1/unit/concrete/Bridgehub/experimental_bridge.t.sol [hunk 4]
```diff
@@ -448,82 +448,30 @@ contract ExperimentalBridgeTest is Test {
...
-    //     if (randomCaller != bridgeOwner) {
-    //         vm.prank(randomCaller);
-    //         vm.expectRevert(bytes("Ownable: caller is not the owner"));
-    //         bridgehub.removeChainTypeManager(randomAddressWithoutTheCorrectInterface);
-    //     }
-    // }
+    function test_removeChainTypeManager_cannotBeCalledByRandomAddress(
+        address _randomAddressWithoutTheCorrectInterface,
+        address _randomCaller
+    ) public {
```
Comment (jcsec-security) [inline l1-contracts/test/foundry/l1/unit/concrete/Bridgehub/experimental_bridge.t.sol:454]: The test case is quite readable, but I don't think it hurts keeping the comments like `// First register our particular chainTypeManage`, `// Not possible to remove by a randomcaller as well` as they make more straightforward to extend the edge cases if necessary.

File: v31-test-followups.md [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+- `testFuzz_regression_bridgeBurnBaseTokenVariousAmounts` (`l1-contracts/test/foundry/l2/integration/L2NativeTokenVaultBridgeBurnRegressionTest.t.sol`): Removed because this zkfoundry wrapper only called `vm.skip(true)`; active coverage is in `L2NativeTokenVaultBridgeBurnRegressionL1Test.t.sol`.
+
+### `l1-contracts/test/unit_tests/l2-upgrade.test.spec.ts` - missing upgrade tx hash log
+
+- Decision: restored as an active test.
+- Reason: the behavior is still relevant: a pending upgrade must force the next committed batch to contain the expected upgrade transaction hash log.
+- Verification target: the test now builds the next batch without the upgrade hash log and asserts `MissingSystemLogs`.
+- Confidence: 98% on the intended assertion and fixture path; local Hardhat execution is blocked because `yarn build` cannot resolve the `forge-std` import from deploy scripts in this checkout, so TypeChain artifacts are not generated.
```
Comment (jcsec-security) [inline v31-test-followups.md]: This is just because the dependencies were not installed before, right?

File: l1-contracts/test/unit_tests/executor_proof.spec.ts [hunk 6]
```diff
@@ -13,113 +12,12 @@ describe("Executor test", function () {
...
-
-    //   const prevCommitment = "0x6ebf945305689a8c3ac993df7f002d41d311a762cd6bf39bb054ead8d1f54404";
-    //   const result = await executor.getBatchProofPublicInput(prevCommitment, nextCommitment, {
-    //     recursionNodeLevelVkHash: "0x5a3ef282b21e12fe1f4438e5bb158fc5060b160559c5158c6389d62d9fe3d080",
-    //     recursionLeafLevelVkHash: "0x72167c43a46cf38875b267d67716edc4563861364a3c03ab7aee73498421e828",
-    //     // ignored.
-    //     recursionCircuitsSetVksHash: "0x05dc05911af0aee6a0950ee36dad423981cf05a58cfdb479109bff3c2262eaac",
-    //   });
-    //   expect(result.toHexString(), "").to.be.equal("0x66876e724acc551e35d48f5c091447a245efcc79d70bb840533ddf83");
+    expect(result.toHexString()).to.equal("0xb29c9adf0177455f74d0a0f38065e77a6d425370d418cd37dbf3eaa0");
```
Comment (jcsec-security) [inline l1-contracts/test/unit_tests/executor_proof.spec.ts]: Not related to the task in hand, but we could move this hex string into a meaningfully named constant

File: system-contracts/test/EvmEmulation.spec.ts (hunk 1 above)
Comment (0xValera) [reply system-contracts/test/EvmEmulation.spec.ts]: I'm not sure, but I don't think so.
The note is there just in case we'll be taking a look and wondering why something is commented out. If however the opcode will become avaialble we should know, and will update the test accordingly.

File: l1-contracts/test/foundry/l1/unit/concrete/Bridgehub/experimental_bridge.t.sol (hunk 4 above)
Comment (0xValera) [reply l1-contracts/test/foundry/l1/unit/concrete/Bridgehub/experimental_bridge.t.sol:454]: added some comments

File: l1-contracts/test/foundry/l1/unit/concrete/Bridgehub/experimental_bridge.t.sol (hunk 3 above)
Comment (0xValera) [reply l1-contracts/test/foundry/l1/unit/concrete/Bridgehub/experimental_bridge.t.sol]: The test was just using obscure naming, I think, I fixed it.
As for CTM, since adding it is an ecosystem operation which is overseen by governance, I think we have guarantees for correct implementation from there.

File: l1-contracts/test/unit_tests/executor_proof.spec.ts (hunk 6 above)
Comment (0xValera) [reply l1-contracts/test/unit_tests/executor_proof.spec.ts]: I addressed it in the scope of this PR

**Review by jcsec-security — APPROVED** [review]

File: v31-test-followups.md [hunk 7]
```diff
@@ -0,0 +1,101 @@
+# v31 Test Coverage Follow-ups
```
Comment (kelemeno) [inline v31-test-followups.md:1]: lets have linear instead. Btw you can integrate linear with the  agents by giving it a linear api key, and include the linear issue number in the PR. its good if you want it comment etc. But lets not have it create issues, it would make a mess
